File: 643. Maximum Average Subarray I/643.maximum-average-subarray-i.py
#
# @lc app=leetcode id=643 lang=python3
#
# [643] Maximum Average Subarray I
#

# @lc code=start
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        current = 0
        for i in range(k):
            current += nums[i]

        answer = current 
        for i in range(k, len(nums)):
            current += nums[i] - nums[i - k]
            answer = max(answer, current)

        return answer / k
# @lc code=end

File: 643. Maximum Average Subarray I/643.maximum-average-subarray-i.c
/*
 * @lc app=leetcode id=643 lang=c
 *
 * [643] Maximum Average Subarray I
 */

// @lc code=start
double findMaxAverage(int* nums, int numsSize, int k) {
    int current = 0;
    for(int i = 0; i < k; i++) 
        current += nums[i];

    int answer = current; 
    for(int i = k; i < numsSize; i++) {
        current += nums[i] - nums[i - k];
        if(answer < current) 
            answer = current;
    }
    return (double)answer / k;
}
// @lc code=end

